pkg/unpack/oci.go

```diff
@@ -159,9 +159,11 @@ func (o OCI) Unpack(ctx context.Context, destination string) (string, error) {
 	return digest.String(), err
 }
 
-func image(ctx context.Context, ref name.Reference, platform containerregistry.Platform, local bool) (containerregistry.Image, error) {
-	if local {
-		return daemon.Image(ref, daemon.WithContext(ctx))
+func fetchImage(ctx context.Context, ref name.Reference, platform containerregistry.Platform, local bool) (containerregistry.Image, error) {
+	// Always attempt to fetch the image locally first.
```

Comment:
This should be configurable IMHO, otherwise you always silently pull a local image without notice. This can be tricky if you do not remember there is an older image in the host...

I'd say local option needs to be explicit. In order to do that I believe the feature we have to implement is adding the `--local` and `--platfrom` (or similar ones) flags in Install/Upgrade/Build etc.

This is in turn something to consider in a separate PR I think, the question here is how to inject this runtime flags, I don't think they belong in Deployment, probably we need a new structure for this sort of runtime data.